### PR TITLE
RDM-1982 Timeout configuration applied to RestTemplate object

### PR DESCRIPTION
**Pull request by tzarsmango**
https://tools.hmcts.net/jira/browse/RDM-1982

Timeout for callbacks were not applied. Added a fix for this and added a timeout test

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/ccd/domain/service/callbacks/CallbackServiceTest.java [hunk 1]
```diff
@@ -82,6 +82,33 @@ public void happyPathWithNoErrorsOrWarnings() throws Exception {
...
+        mockCallbackServer.stubFor(post(urlMatching("/test-callback.*"))
+            .willReturn(okJson(mapper.writeValueAsString(callbackResponse)).withStatus(200).withFixedDelay(1500)));
+
+        long startTime = System.nanoTime();
+        final Optional<CallbackResponse> result = callbackService.send(testUrl, null, caseEvent, caseDetails);
+        long endTime = System.nanoTime();
+
+        long duration = (endTime - startTime);
+        final CallbackResponse response = result.orElseThrow(() -> new AssertionError("Missing result"));
+        assertTrue(duration/1000000 > 2500);
```
Comment (mario-paniccia) [inline src/test/java/uk/gov/hmcts/ccd/domain/service/callbacks/CallbackServiceTest.java]: can you use 1__000__000 ? helps readability

Comment (mario-paniccia) [reply src/test/java/uk/gov/hmcts/ccd/domain/service/callbacks/CallbackServiceTest.java]: if we are testing retry, can we do a kind of verify with wiremock to make sure the endpoint is called twice? not sure whether wiremock supports that

Comment (tzarsmango) [reply src/test/java/uk/gov/hmcts/ccd/domain/service/callbacks/CallbackServiceTest.java]: both comments appreciated and applied 👍

File: src/main/java/uk/gov/hmcts/ccd/domain/service/callbacks/CallbackService.java [hunk 2]
```diff
@@ -119,6 +119,7 @@ private Integer secondsToMilliseconds(final Integer seconds) {
             requestFactory.setConnectionRequestTimeout(secondsToMilliseconds(timeout));
             requestFactory.setReadTimeout(secondsToMilliseconds(timeout));
             requestFactory.setConnectTimeout(secondsToMilliseconds(timeout));
+            restTemplate.setRequestFactory(requestFactory);
             return Optional.ofNullable(
```
Comment (adr1ancho) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/callbacks/CallbackService.java]: I am now confused.  A while back, restTemplate was created inside this function.  Then it was injected.  Now we are changing the properties of this restTemplate.  Should we create a restTemplate as per call back??  Just a thought

Comment (tzarsmango) [reply src/main/java/uk/gov/hmcts/ccd/domain/service/callbacks/CallbackService.java]: Your point may be right, but I think we should discuss it out of the scope of this bug fix. Perhaps with another task to validate the need for that.
Just for the sake of completing this task.

**Pull request merged**